Tests/StreamChatTests/Database/DTOs/ChannelReadDTO_Tests.swift

```diff
@@ -272,7 +272,103 @@ final class ChannelReadDTO_Tests: XCTestCase {
         XCTAssertTrue(observer.updatedMessageIDs.isEmpty)
     }
 
-    // MARK: - markChannelAsUnread
+    // MARK: - markChannelAsUnread - partial
+
+    func test_markChannelAsUnreadPartial_whenReadDoesNotExist() throws {
+        // GIVEN
+        let cid = ChannelId.unique
+        let userId = UserId.unique
+        let messageId = MessageId.unique
+
+        // WHEN
+        try database.writeSynchronously { session in
+            session.markChannelAsUnread(for: cid, userId: userId, from: messageId)
+        }
+
+        // THEN
+        XCTAssertEqual(database.writeSessionCounter, 1)
+        XCTAssertNil(readDTO(cid: cid, userId: userId))
+    }
+
+    func test_markChannelAsUnreadPartial_whenMessageDoesNotExist() throws {
+        // GIVEN
+        let cid = ChannelId.unique
+        let userId = UserId.unique
+        let messageId = MessageId.unique
+
+        let member: MemberPayload = .dummy(user: .dummy(userId: userId))
+        let read = ChannelReadPayload(
+            user: member.user!,
+            lastReadAt: .init(),
+            unreadMessagesCount: 10
+        )
+
+        let channel: ChannelPayload = .dummy(
+            channel: .dummy(cid: cid),
+            members: [member],
+            channelReads: [read]
+        )
+
+        try database.writeSynchronously { session in
+            try session.saveChannel(payload: channel)
+        }
+
+        database.writeSessionCounter = 0
+
+        // WHEN
+        try database.writeSynchronously { session in
+            session.markChannelAsUnread(for: cid, userId: userId, from: messageId)
+        }
+
+        // THEN
+        XCTAssertEqual(database.writeSessionCounter, 1)
+        XCTAssertNotNil(readDTO(cid: cid, userId: userId))
```

Comment:
Do we want to check here the `unreadCount` to ensure that nothing is being updated?